### PR TITLE
TEST: Added Unit tests for campus delete method

### PR DESCRIPTION
**Pull request by SamAnast**
# Description

Added unit test for campus delete method

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: test/api/campuses_test.rb [hunk 1]
```diff
@@ -52,4 +52,52 @@ def test_put_campuses
...
+    remove_property(Project)
+    remove_property(Tutorial)
+
+    # perform the delete
+    delete_json with_auth_token("/api/campuses/#{campus_id}", user)
+
+    assert_equal 403, last_response.status
+  end
+
+  #This method required to test campus delete methods
```
Comment (macite) [inline test/api/campuses_test.rb]: Can you also describe what it does... it is not clear from the name.

Maybe consider a better name - something like delete all associated with campus

File: test/api/campuses_test.rb [hunk 2]
```diff
@@ -52,4 +52,52 @@ def test_put_campuses
...
+  def remove_property(prop)
```
Comment (macite) [inline test/api/campuses_test.rb]: Also add a campus id property rather than using a hard coded 1 here

File: test/api/campuses_test.rb [hunk 3]
```diff
@@ -52,4 +52,52 @@ def test_put_campuses
...
+      prop_array = prop.all  
+      n = prop_array.length
+    
+      for i in 0..n-1 do
+        campus_id = prop_array[i].campus.id
+
+        if campus_id == 1
```
Comment (macite) [inline test/api/campuses_test.rb]: use parameter here rather than 1

File: test/api/campuses_test.rb [hunk 4]
```diff
@@ -52,4 +52,52 @@ def test_put_campuses
     assert_json_matches_model(last_response_body, first_campus, response_keys)
     assert_equal 0, first_campus[:mode]
   end
+    
+  def test_delete_campuses    
+    initial_num_of_campus = Campus.all.count
+    user = User.admins.first
+
+    remove_property(Project)
```
Comment (macite) [inline test/api/campuses_test.rb]: Rename - see related comments below...

File: test/api/campuses_test.rb [hunk 5]
```diff
@@ -37,6 +45,15 @@ def test_post_campuses
     assert_equal 0, campus[:mode]
   end
 
+  def test_student_post_campuses
+    user_student = FactoryGirl.create(:user, :student)
+    data_to_post = { campus: FactoryGirl.build(:campus, mode: 'timetable') }
+
+    post_json  with_auth_token("/api/campuses", user_student), data_to_post
+    assert_equal 403, last_response.status
```
Comment (macite) [inline test/api/campuses_test.rb]: Add checks that the created campus matches the data you sent through

File: test/api/campuses_test.rb [hunk 6]
```diff
@@ -52,4 +69,40 @@ def test_put_campuses
     assert_json_matches_model(last_response_body, first_campus, response_keys)
     assert_equal 0, first_campus[:mode]
   end
+
+  def test_student_put_campuses
+    user_student = FactoryGirl.create(:user, :student)
+    data_to_put ={ campus: FactoryGirl.build(:campus, mode: 'timetable') }
+
+    # Update campus with id = 1
+    put_json with_auth_token("/api/campuses/1", user_student), data_to_put
```
Comment (macite) [inline test/api/campuses_test.rb]: Do not rely upon the campus existing... use factory girl to create

File: test/api/campuses_test.rb [hunk 7]
```diff
@@ -52,4 +69,40 @@ def test_put_campuses
...
+
+    #Perform the delete
+    delete_json with_auth_token("/api/campuses/#{campus.id}")
+    #check the request went through
+    assert_equal 200, last_response.status
+
+    get '/api/campuses' #get number of campuses
+    # check if current number of campuses = original number of campuses
+    assert_equal initial_num_of_campus - 1, last_response_body.count
+  end
```
Comment (macite) [inline test/api/campuses_test.rb]: You can also check that the campus no longer exists here if you query for it.

File: test/factories/campuses_factory.rb [hunk 8]
```diff
@@ -1,5 +1,6 @@
 FactoryGirl.define do
   factory :campus do
+    sequence(:id, (4..50).cycle) { |n| n }
```
Comment (macite) [inline test/factories/campuses_factory.rb]: id is automatically assigned. Remove this from FactoryGirl.

Comment (macite) [reply test/factories/campuses_factory.rb]: This is most likely masking some other issue with the tests.

File: test/api/campuses_test.rb (hunk 5 above)
Comment (SamAnast) [reply test/api/campuses_test.rb]: Sorry for the confusion. This method is designed to test that students lack auth to put data.
I've updated the method name to reflect this more clearly.

File: test/api/campuses_test.rb [hunk 9]
```diff
@@ -52,4 +69,42 @@ def test_put_campuses
...
     assert_equal 0, first_campus[:mode]
   end
+
+  def test_student_cannot_put_campuses
+    user_student = FactoryBot.build(:user, :student)
+    data_to_put ={ campus: FactoryBot.create(:campus, mode: 'timetable')}
+
+    put_json with_auth_token("/api/campuses/#{data_to_put[:campus].id}", user_student), data_to_put
+    assert_equal 403, last_response.status
+
```
Comment (macite) [inline test/api/campuses_test.rb]: Avoid non-purposeful blank lines... such as this one.

Comment (macite) [reply test/api/campuses_test.rb]: Make sure to review your changes before you commit.

File: test/api/campuses_test.rb [hunk 10]
```diff
@@ -1,5 +1,6 @@
 require 'test_helper'
 
+
```
Comment (macite) [inline test/api/campuses_test.rb]: Avoid non-purposeful blank lines... such as this one.

File: test/api/campuses_test.rb [hunk 11]
```diff
@@ -37,6 +45,15 @@ def test_post_campuses
     assert_equal 0, campus[:mode]
   end
 
+  def test_student_cannot_post_campuses
+    user_student = FactoryBot.create(:user, :student)
+    data_to_post = { campus: FactoryBot.build(:campus, mode: 'timetable') }
+
+    post_json  with_auth_token("/api/campuses", user_student), data_to_post
+    assert_equal 403, last_response.status
+
```
Comment (macite) [inline test/api/campuses_test.rb]: Avoid non-purposeful blank lines... such as this one.

File: test/api/campuses_test.rb [hunk 12]
```diff
@@ -52,4 +69,42 @@ def test_put_campuses
...
+    put_json with_auth_token("/api/campuses/#{data_to_put[:campus].id}", user_student), data_to_put
+    assert_equal 403, last_response.status
+ 
+  end
+    
+  def test_delete_campuses        
+    campus = FactoryBot.create(:campus, mode: 'timetable')
+    initial_num_of_campus = Campus.all.count
+
+
```
Comment (macite) [inline test/api/campuses_test.rb]: ...

File: test/factories/campuses_factory.rb (hunk 8 above)
Comment (macite) [reply test/factories/campuses_factory.rb]: This should not be here! We do not assign ids.

**Pull request merged**